roc.go

```diff
@@ -50,22 +50,36 @@ func ROC(n int, y []float64, classes []bool, weights []float64) (tpr, fpr []floa
 		if n < 2 {
 			panic("stat: cannot calculate fewer than 2 points on a ROC curve")
 		}
+		tpr = make([]float64, n)
+		fpr = make([]float64, n)
+		if len(y) == 0 {
+			return
+		}
 		incWidth = (y[len(y)-1] - y[0]) / float64(n-1)
-		if incWidth <= 0 {
-			panic("stat: input constant")
+		if incWidth == 0 {
+			for _, u := range classes {
+				if u {
+					tpr[n-1] = 1
+				} else {
+					fpr[n-1] = 1
+				}
+			}
+			return
```

Comment:
There has to be a better way to do this. Essentially I want 
```
if any(classes) {
  tpr[n-1]=1
}
if any(!classes){
  fpr[n-1]=1
}
```
I just don't know how to do `any' in go